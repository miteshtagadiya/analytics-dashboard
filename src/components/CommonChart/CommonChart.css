.container-CommonChart {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
  padding: 10px 10px 0px 10px;
  width: 100%;
  height: 270px;
}
.container-CommonChart .recharts-legend-wrapper {
  position: absolute;
  left: 0px !important;
}

.recharts-text .recharts-cartesian-axis-tick-value {
  font-weight: bold !important;
  color: #404b69 !important;
}
.container-CommonChart .recharts-default-legend {
  margin-bottom: 10px !important;
}
.recharts-legend-item-text {
  color: #404b69;
  font-size: 14px !important;
}

.recharts-legend-item {
  min-width: 90px;
}
.container-CommonChart .recharts-surface {
  margin-left: -30px !important;
}

.container-CommonChart .no-data {
  height: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  font-size: 11px;
  color: black;
}

.recharts-cartesian-axis {
  font-family: "Work Sans" !important;
  font-size: 11px !important;
  color: #404b69 !important;
  font-weight: bold !important;
}
.recharts-tooltip-label {
  color: #404b69 !important;
}
