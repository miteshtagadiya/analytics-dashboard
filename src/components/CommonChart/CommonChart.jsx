import React from "react";
import {
  LineChart,
  BarChart,
  Line,
  Bar,
  XAxis,
  YAxis,
  Tooltip,
  Legend,
  ResponsiveContainer,
  CartesianGrid,
} from "recharts";
import "./CommonChart.css";

const CommonChart = (props) => {
  const data = props.data === undefined ? [] : props.data;

  const CustomTooltip = ({ active, payload, label }) => {
    if (active) {
      return (
        <div
          className="custom-tooltip"
          style={{
            background: "white",
            padding: 15,
            borderRadius: 10,
            border: "1px solid grey",
          }}
        >
          <p
            style={{
              textAlign: "center",
              fontWeight: "bold",
              color: "#404b69",
            }}
          >
            {label}
          </p>
          {typeof payload !== "undefined" &&
            payload !== null &&
            payload.map((item, index) => {
              return (
                <div key={index}>
                  <span style={{ color: item.color, fontWeight: "bold" }}>
                    {item.name}:{" "}
                  </span>
                  <span style={{ color: item.color, fontWeight: "bold" }}>
                    {item.value}
                  </span>
                  <br />
                </div>
              );
            })}
        </div>
      );
    }

    return null;
  };

  let ChartName = (chart) => {
    switch (chart) {
      case "LineChart":
        return LineChart;
      case "BarChart":
        return BarChart;
      default:
        return LineChart;
    }
  };

  let selectChart = (chart) => {
    switch (chart) {
      case "LineChart":
        return Line;
      case "BarChart":
        return Bar;
      default:
        return Line;
    }
  };

  let Chart = selectChart(props.chart);
  let Type = ChartName(props.chart);
  return (
    <div
      className="container-CommonChart"
      style={props.height ? { height: props.height } : { height: 270 }}
    >
      {data.length > 0 ? (
        <ResponsiveContainer width="100%" height="100%">
          <Type width={500} height={100} data={data}>
            <CartesianGrid strokeDasharray="3 3" />

            <XAxis hide={false} dataKey={props.axis ? props.axis : "name"} />

            <YAxis />
            <Tooltip content={<CustomTooltip />} />
            <Legend
              layout={props.legendLayout ? props.legendLayout : "horizontal"}
              margin={{ top: 0, left: 0, right: 0, bottom: 10 }}
              verticalAlign={
                props.verticalAlign ? props.verticalAlign : "bottom"
              }
            />
            {props.labels &&
              props.labels.map((label, index) =>
                props.chart === "BarChart" ? (
                  <Chart
                    key={index}
                    stackId={"a"}
                    type="monotone"
                    dataKey={label}
                    stroke={props.colors[index]}
                    fill={props.colors[index]}
                  />
                ) : (
                  <Chart
                    key={index}
                    stackId={"a"}
                    type="monotone"
                    dataKey={label}
                    stroke={props.colors[index]}
                    activeDot={{ r: 8 }}
                    fill={props.colors[index]}
                    dot={false}
                  />
                )
              )}
          </Type>
        </ResponsiveContainer>
      ) : (
        <div className="no-data">No data</div>
      )}
    </div>
  );
};

export default CommonChart;
