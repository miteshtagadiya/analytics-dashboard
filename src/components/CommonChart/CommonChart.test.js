import React from "react";
import renderer from "react-test-renderer";
import CommonChart from "./CommonChart";

test("renders correctly when there is no props", () => {
  const tree = renderer.create(<CommonChart />).toJSON();
  expect(tree).toMatchSnapshot();
});
