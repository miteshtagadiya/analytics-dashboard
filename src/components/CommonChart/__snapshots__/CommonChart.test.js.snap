// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is no props 1`] = `
<div
  className="container-CommonChart"
  style={
    Object {
      "height": 270,
    }
  }
>
  <div
    className="no-data"
  >
    No data
  </div>
</div>
`;
